Feature/2828 input styles

emclaug2 opened the pull request:
<!-- If this pull request fixes an Issue, link it below. If not, you can remove the line below -->
Fixes #48 

<!-- Include a bulleted list summarizing the main changes you have made in this PR -->
#### Changes proposed in this Pull Request:
- Does not address no. 7 in the list yet but I would like to get some eyes on this.

<!-- Instruction for PR reviewers, if more complicated than a simple yarn start -->
#### To Test:
- Run the showcase

jeffvg commented:
are the style updates for Standard Text Field and Filled Text Field too? I can see the outlined is almost 52px

![image](https://user-images.githubusercontent.com/37150565/152048706-71da572c-e841-408f-91c8-1aa17a4dfbd9.png)

emclaug2 commented:
> are the style updates for Standard Text Field and Filled Text Field too? I can see the outlined is almost 52px

I've been using the `mat-form-field-flex` element as the parent el for measuring the height of these input elements.  
![image](https://user-images.githubusercontent.com/6538289/152049482-8c29d51d-dda5-45bf-933c-e589380837ce.png)

emclaug2 commented on _common.scss:
```diff
@@ -144,4 +154,69 @@
+        }
+
+        &.mat-form-field-appearance-standard {
+            &.mat-form-field-should-float .mat-form-field-label {
```
I'm thinking that these should be rem since they deal with vertical margins.

huayunh commented:
- [ ] Resolve conflict
- [ ] Update the Angular showcase demo repo and submodule links
- [ ] trailing and **leading** icons shall be vertically centered. So the alignment shall be applied on `matPrefix` icons too. 
- [ ] prefixed icons should be textSecondary, right now they are textPrimary
- [ ] prefixed icons' left spacing should stay 16; suffixed icons' right spacing should be 12
![image](https://user-images.githubusercontent.com/8997218/152220082-aed5c661-7af3-4375-848f-c58f0ba4afd7.png)
- [ ] `.mat-form-field-wrapper` should have a `padding-bottom` of `20px` / `1.2rem` 
- [ ] The legacy form field feels "taller" than the standard form field (the suffix icon position feels different)
![image](https://user-images.githubusercontent.com/8997218/152221056-17b1bb3c-7754-4f9c-968d-7503da4c2916.png)
- [ ] Remember to test your styling at 2x font size. This is what it currently looks like with chrome set to "extra large" font
![image](https://user-images.githubusercontent.com/8997218/152221388-62a84130-2f45-44d3-8799-3e55e45b00d0.png)

emclaug2 commented:
Thanks for the in-depth review @huayunh , I'm going to update the angular showcase demo to use suffix and prefix icons so the testing will be a little easier.

emclaug2 commented:
@huayunh I'm thinking this is ready for another round of review; I ran into some issues applying a `1.2 rem` padding to the class you specified since the legacy and standard variants were starting to look off.  I don't think the styles are still 100% centered but I could use a second set of eyes on this for sure.

The demo has been updated and hosted here: https://pxb-angular-theme-demo.web.app/material-components/input-components

emclaug2 commented:
Normal:

![image](https://user-images.githubusercontent.com/6538289/152386121-61a2830c-7556-434f-be67-93ef835b3123.png)


Xtra Large: 

![image](https://user-images.githubusercontent.com/6538289/152386407-4ca430f8-2371-4ced-9fe5-29100acfa2d6.png)

huayunh commented:
Thanks for the firebase demo. This looks awesome!

# 1 
Right to left, the horizontal padding seems backward for filled text fields. (Outlined text fields are OK)
![image](https://user-images.githubusercontent.com/8997218/152399322-06d0af50-b19a-4ff4-92ed-a5ff904e63ec.png)

# 2
The comment about `.mat-form-field-wrapper` was just to ensure that this green region is 20px tall (No.4 bullet point in the original issue). The difference is relatively minor and I find it already a pretty good improvement compared to the original state.
![image](https://user-images.githubusercontent.com/8997218/152400110-001ac1e7-6f75-42e1-9ade-e138a41eb574.png)

# 3
This is probably a result of my eyes being hyper-sensitive. The vertical spacing feels 1px high but... nah maybe I will be the only person reacting to it.
![image](https://user-images.githubusercontent.com/8997218/152401032-e3732fda-2e3b-4252-b67d-f5fed09cd325.png)

# 4
Right to left + 2x font size is off
![image](https://user-images.githubusercontent.com/8997218/152401489-23bce9a4-3d6f-42ff-8316-5496abdb92a2.png)

huayunh commented:
(Don't spend too much effort on the third comment from above)

emclaug2 commented:
> # 4
> Right to left + 2x font size is off 

@huayunh Can you clarify what's off about the font-size?

emclaug2 commented:
**Before:**

2X
![Large](https://user-images.githubusercontent.com/6538289/152413795-6341f4e3-f374-4a00-bd7b-aeffe3b3325a.PNG)

1X
![Normal](https://user-images.githubusercontent.com/6538289/152413797-f969bc18-66f7-44a3-8992-dedbf2332bdc.PNG)

**After:**

2X
![image](https://user-images.githubusercontent.com/6538289/152414008-d93f324c-52c7-4341-b48a-dd3826fe821c.png)

1X
![image](https://user-images.githubusercontent.com/6538289/152413881-0ad49f5b-b62a-49d2-b238-4d1b0d4b7d06.png)

huayunh reviewed (approved):
Eyeballing looks good. 

I saw a mix of `em` and `rem` in the code; I assume you have you very specific reason for doing it this way.

emclaug2 commented:
> Eyeballing looks good.
> 
> I saw a mix of `em` and `rem` in the code; I assume you have you very specific reason for doing it this way.

Great! I went back and removed those lingering `em`s.  There wasn't a reason to keep them in & I've replaced them with `rem`.  Running through the demo one last time, it looks the same.

huayunh commented:
I just noticed... in the dark theme, the filled warning / required text field has a different background color.

![image](https://user-images.githubusercontent.com/8997218/152567598-9aa50fd2-beb1-4b19-ae91-f1e0f0ce152e.png)
